Core/GDCore/IDE/WholeProjectRefactorer.cpp

```diff
@@ -1382,25 +1382,12 @@ void WholeProjectRefactorer::ObjectOrGroupRemovedInLayout(
 
   // Remove object in external events
   if (removeEventsAndGroups) {
-    DependenciesAnalyzer analyzer(project, layout);
-    if (analyzer.Analyze()) {
-      for (auto& externalEventsName :
-           analyzer.GetExternalEventsDependencies()) {
-        auto& externalEvents = project.GetExternalEvents(externalEventsName);
-        gd::EventsRefactorer::RemoveObjectInEvents(project.GetCurrentPlatform(),
-                                                   project,
-                                                   layout,
-                                                   externalEvents.GetEvents(),
-                                                   objectName);
-      }
-      for (auto& layoutName : analyzer.GetScenesDependencies()) {
-        auto& layout = project.GetLayout(layoutName);
-        gd::EventsRefactorer::RemoveObjectInEvents(project.GetCurrentPlatform(),
-                                                   project,
-                                                   layout,
-                                                   layout.GetEvents(),
-                                                   objectName);
-      }
+    for (auto &externalEventsName :
+         GetAssociatedExternalEvents(project, layout.GetName())) {
```

Comment:
In the description of the function in the .h, change 

```
   * This will update the layout, all external layouts associated with it
   * and all external events used by the layout.
```

by:

```
   * This will update the layout, all external layouts associated with it
   * and all external events associated with it.
```

both for ObjectOrGroupRenamedInLayout and ObjectOrGroupRemovedInLayout